### PR TITLE
docs: add 3.12.0 upgrade note with ApiKey migration script for mongoDB 

### PR DESCRIPTION
**Pull request by marcambier**
https://github.com/gravitee-io/issues/issues/6005

**WARN : do not merge this issue before step2 is done (https://github.com/gravitee-io/issues/issues/6006)
(you can actually review it and test non regression) I'll PR step2 branch into this one**

### CODE REVIEW

File: upgrades/3.x/3.12.0/README.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+= Upgrade to 3.12.0
+
+== Model
+
+=== Api keys
+
+Before this version, ApiKey contains a *key* attribute, which is the value of the key, and at the same time his database id.
```
Comment (phiz71) [inline upgrades/3.x/3.12.0/README.adoc:7]:
```suggestion
Before this version, ApiKey contained a *key* attribute, which is the value of the key, and at the same time his database id.
```

File: upgrades/3.x/README.adoc [hunk 2]
```diff
@@ -1,6 +1,8 @@
 = Upgrade Notes
 
 ifdef::env-github[]
+include::3.12.0/README.adoc[leveloffset=+1]
```
Comment (phiz71) [inline upgrades/3.x/README.adoc:4]: Not necessary in this PR, will be done with https://github.com/gravitee-io/release/pull/337

File: upgrades/3.x/README.adoc [hunk 3]
```diff
@@ -58,6 +60,10 @@ You may be required to perform manual actions as part of the upgrade.
 WARNING: Be sure to run scripts on the correct database since `gravitee` is not always the default database!
 Check your db name by running `show dbs;`
 
+include::https://raw.githubusercontent.com/gravitee-io/release/master/upgrades/3.x/3.12.0/README.adoc[leveloffset=+1]
```
Comment (phiz71) [inline upgrades/3.x/README.adoc:63]: Not necessary in this PR, will be done with https://github.com/gravitee-io/release/pull/337

File: upgrades/3.x/README.adoc [hunk 4]
```diff
@@ -58,6 +60,10 @@ You may be required to perform manual actions as part of the upgrade.
...
+
+include::https://raw.githubusercontent.com/gravitee-io/release/master/upgrades/3.x/3.11.0/README.adoc[leveloffset=+1]
```
Comment (phiz71) [inline upgrades/3.x/README.adoc:65]: Not necessary in this PR, it has already been fixed.

**Review by phiz71 — CHANGES_REQUESTED** [review]

File: upgrades/3.x/3.12.0/README.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+Those 3 Portal API's endpoints now expose distinct *id* and *key* in HTTP response (previously, the *key* attribute was exposed as '*id*') :
+
+- GET /subscriptions/{subscriptionId}
+- POST /subscriptions/{subscriptionId}
+- POST /subscriptions/{subscriptionId}/_renew
+
+== Repository
+
+=== Mongodb
```
Comment (tiffanymarchi) [inline upgrades/3.x/3.12.0/README.adoc:22]: Can we please change to: 

Starting in version 3.12.0, APIKeys now contain distinct attributes: 

The 3 Portal API endpoints listed below now expose distinct *ID* and *key* in the HTTP response (previously, the *key* attribute was exposed as '*ID*'):

=== MongoDB

**Review by tiffanymarchi — COMMENTED** [review]
Hey @marcambier -- nicely done! Can you please just make the small edits for formatting/grammar? Thank you!!

File: upgrades/3.x/README.adoc (hunk 2 above)
Comment (marcambier) [reply upgrades/3.x/README.adoc]: ok, removed it

File: upgrades/3.x/README.adoc (hunk 3 above)
Comment (marcambier) [reply upgrades/3.x/README.adoc]: ok, removed it

File: upgrades/3.x/README.adoc (hunk 4 above)
Comment (marcambier) [reply upgrades/3.x/README.adoc]: ok, removed it

File: upgrades/3.x/3.12.0/README.adoc [hunk 6]
```diff
@@ -0,0 +1,27 @@
+= Upgrade to 3.12.0
+
+== Model
+
+=== Api keys
+
```
[the remark this answers is not in the document]
Comment (marcambier) [reply upgrades/3.x/3.12.0/README.adoc]: :+1:  done

File: upgrades/3.x/3.12.0/README.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+Before this version, ApiKey contained a *key* attribute, which is the value of the key, and at the same time his database id.
```
[the remark this answers is not in the document]
Comment (marcambier) [reply upgrades/3.x/3.12.0/README.adoc]: +1 done

File: upgrades/3.x/3.12.0/README.adoc (hunk 5 above)
Comment (marcambier) [reply upgrades/3.x/3.12.0/README.adoc]: :+1: done

File: upgrades/3.x/3.12.0/mongodb/api-keys-migration.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+print('Add \'key\' and \'api\' columns in \'keys\' table');
```
Comment (gcusnieux) [inline upgrades/3.x/3.12.0/mongodb/api-keys-migration.js:1]: You can use the `Template literals` to avoid having to escape

Comment (marcambier) [reply upgrades/3.x/3.12.0/mongodb/api-keys-migration.js:1]: good to know, i changed it

**Review by ytvnr — APPROVED** [review]

**Pull request merged**